src/LibrarianDao.java

```diff
@@ -0,0 +1,47 @@
+import java.sql.*;
+public class LibrarianDao {
+
+	
+	public static int save(String name,String password,String email,String address,String city,String contact){
+		int status=0;
+		try{
+			Connection con=DB.getConnection();
+			PreparedStatement ps=con.prepareStatement("insert into librarian(name,password,email,address,city,contact) values(?,?,?,?,?,?)");

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *ps*. The resource is referenced in statements at the following lines: 9, 10, 11, 12, 13, 14, 15, 16. The resource closure statement is at line: 17. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by PreparedStatement.setString. Either a) close *ps* in a try-finally block or b) close the resource by declaring *ps* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).